### ISSUE TITLE
Add iOS support

### ISSUE THREAD

**Issue by Jerakin** [andsve/def-diags#6]
Maybe this is relevant

https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/UsingtheOpenandSavePanels/UsingtheOpenandSavePanels.html#//apple_ref/doc/uid/TP40010672-CH4-SW1

**Comment by andsve** [comment]
Really same answer as in #7, looks like it should be doable. Need to investigate.

**Comment by andsveking** [comment]
Did a little investigation on this, sadly not sure it's a good fit for def-diags. One reason is that the linked URL is aimed for macOS (AppKit), but there is a similar iOS API under UIKit using `UIDocumentPickerViewController` but still doesn't fit into the synchronous path/URI API that def-diags aims to provide. It might be possible to shoehorn in, but maybe a separate mobile extension for handling document access is a better option?

**Comment by Jerakin** [comment]
Yeah, if it doesn't fit well with it then I think a new extension could be better
